packages/astro/src/actions/runtime/route.ts

```diff
@@ -0,0 +1,41 @@
+import type { APIRoute } from '../../@types/astro.js';
+import { ApiContextStorage } from './store.js';
+import { formContentTypes, getAction } from './utils.js';
+import { ActionError } from './virtual/shared.js';
+
+export const POST: APIRoute = async (context) => {
+	const { request, url } = context;
+	if (request.method !== 'POST') {
+		return new Response(null, { status: 405 });
+	}
+	const actionPathKeys = url.pathname.replace('/_actions/', '').split('.');
+	const action = await getAction(actionPathKeys);
+	const contentType = request.headers.get('Content-Type');
+	let args: unknown;
+	if (formContentTypes.some((f) => contentType?.startsWith(f))) {
+		args = await request.clone().formData();
+	} else if (contentType === 'application/json') {
+		args = await request.clone().json();
+	} else {
+		return new Response(null, { status: 415 });
```

Comment:
nit: maybe let's add the MDN link to status code 415

Reply:
Ah, in the Response body or as a comment?

Reply:
Pretty sure he means in a comment.

Reply:
Yeah just a comment. 415 isn't a very common code that you see around, so a link would be helpful